apps/ui/src/components/user/EditUserCard.vue

```diff
@@ -1,289 +1,307 @@
 
-sourceData.onResult(reloadInputData);
-onMounted(reloadInputData);
 
-const editor = ref<UserPermissionsEditor | null>(null);
-const submitting = ref<boolean>(false);
-const loadingError = ref<string>("");
-const inputUser = ref<Partial<User>>({});
-const userGroupsToBeDeleted = ref<UserGroup[]>([]);
-const groupsToBeAdded = ref<Pick<Group, "name" | "id">[]>([]);
-const userDetailsInput = ref<UserDetailsInput | null>(null);
-const showPermissionsEditor = ref<boolean>(false);
+sourceData.onResult((result) => {
+  if (result.loading) return;
+  const user = result.data.user;
 
-const confirmPermissionsEditor = computed({
-  get: () => showPermissionsEditor.value,
-  set: (value: boolean) => {
-    if (!value && editor.value.hasUnsavedChanges) {
-      dialog.warning({
-        title: "Close without saving?",
-        content: "Are you sure you want to close without saving?",
-        positiveText: "Yes",
-        negativeText: "Go Back",
-        onPositiveClick: () => {
-          showPermissionsEditor.value = false;
-        },
-      });
-    } else {
-      showPermissionsEditor.value = value;
-    }
-  },
-});
+  oldGroups.value = [];
+  newGroups.value = [];
+  oldPermissions.value = [];
+  newPermissions.value = [];
 
-const currentUserGroups = computed<UserGroup[]>(() => {
-  if (!sourceData.result.value?.user?.groups) {
-    return [];
+  userData.value = {
+    username: user?.username,
+    mail: user?.mail,
+    discord: user?.discord,
   }
-  return sourceData.result.value.user?.groups?.filter(
-    (ug) => !userGroupsToBeDeleted.value.find((ug2) => ug2.id === ug.id)
-  );
-});
 
-function reloadInputData() {
-  loadingError.value = "";
-  const result = sourceData.result.value;
-  if (!result?.user) {
-    loadingError.value = "User not found";
-  }
-  inputUser.value = { ...result?.user };
-  groupsToBeAdded.value = [];
-  userGroupsToBeDeleted.value = [];
-}
+  const groups = user?.groups;
+  if (groups)
+    for (let i = 0; i < groups.length; i++) {
+      oldGroups.value[i] = groups[i];
+      newGroups.value[i] = groups[i].group as Group;
+    }
+
+
+  const permissions = user?.permissions;
+  if (permissions)
+    for (let i = 0; i < permissions.length; i++) {
+      oldPermissions.value[i] = permissions[i];
+      newPermissions.value[i] = permissions[i];
+    }
+
 
-async function save() {
-  if (!userDetailsInput.value?.isValid) {
+  const personResult = user?.person;
+  if (personResult)
+    person.value = personResult;
+})
+
+async function validate(next: () => void) {
+  loading.value = false;
+  if (step.value < 5) {
+    if (validation.value)
+      next();
     return;
   }
-  submitting.value = true;
+  try {
+    await updateUserMutation.mutate({
+      id: props.id,
+      data: {
+        username: userData.value.username,
+        mail: userData.value.mail,
+        discord: userData.value.discord,
+        personId: person.value.id ?? null
```

Comment:
When the person is removed, `PeopleRow` sets the person ID to 0. `0 ?? null` still returns 0. Submitting this to the API results in a database error, since person ID doesn't exist. Null needs to be provided to the API explicitly, so it knows to remove the person instead of trying to link a new one. Consider using `person.value.id || null` instead.